scripts/3.TCGA-MLexample_Pathway.py

```diff
@@ -254,89 +238,69 @@ def fs_mad(x, y):
     scores = mad(x) 
     return scores, np.array([np.NaN]*len(scores))
 
-# select the top features with the highest MAD
-feature_select = SelectKBest(fs_mad, k=n_feature_kept)
-
 
 # ## Define pipeline and Cross validation model fitting
 
-# In[28]:
-
-# Include loss='log' in param_grid doesn't work with pipeline somehow
-clf = SGDClassifier(random_state=0, class_weight='balanced',
-                    loss=param_fixed['loss'], penalty=param_fixed['penalty'])
+# In[27]:
 
-# joblib is used to cross-validate in parallel by setting `n_jobs=-1` in GridSearchCV
-# Supress joblib warning. See https://github.com/scikit-learn/scikit-learn/issues/6370
-warnings.filterwarnings('ignore', message='Changing the shape of non-C contiguous array')
-clf_grid = GridSearchCV(estimator=clf, param_grid=param_grid, n_jobs=-1, scoring='roc_auc')
-pipeline = make_pipeline(
-    feature_select,  # Feature selection
-    StandardScaler(),  # Feature scaling
-    clf_grid)
+# Parameter Sweep for Hyperparameters
+param_grid = {
+    'select__k': [8000],
+    'classify__loss': ['log'],
+    'classify__penalty': ['elasticnet'],
+    'classify__alpha': [10 ** x for x in range(-3, 1)],
+    'classify__l1_ratio': [0, 0.2, 0.8, 1],
+}
 
+pipeline = Pipeline(steps=[
+    ('select', SelectKBest(fs_mad)),
+    ('standardize', StandardScaler()),
+    ('classify', SGDClassifier(random_state=0, class_weight='balanced'))
+])
 
-# In[29]:
+cv_pipeline = GridSearchCV(estimator=pipeline, param_grid=param_grid, n_jobs=-1, scoring='roc_auc')
```

Comment:
so that I'm clear, the `GridSearchCV` will build the pipeline according to the steps defined on line 255? So that each step is performed separately in each fold. E.g. `SelectKBest` is performed on each cross validation training fold independently?